### PR TITLE
feature added

### PR DESCRIPTION
**Pull request by ihorandrianov**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,102 @@
...
+      + `${this.chipVersion}, Weight: ${this.weight}`;
+  }
 
-class DeliveryDrone {}
+  goForward(y = 1) {
+    if (y > 0) {
+      this.coords.y += y;
+    }
+
+    return this;
```
Comment (DenysKleimenov) [inline src/robotFactory.js:25]: We don't need to return the object, just to change the coordinates inside of the object,

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,102 @@
...
+    this.coords = {
+      x: 0,
+      y: 0,
+      z: 0,
+      ...coords,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  goUp(z = 1) {
```
Comment (DenysKleimenov) [inline src/robotFactory.js:63]: Name 'step' is clearer

File: src/robotFactory.js [hunk 3]
```diff
@@ -1,10 +1,102 @@
...
+    super(name, weight, coords, chipVersion);
+    this.currentLoad = currentLoad;
+    this.maxLoadWeight = maxLoadWeight;
+  }
+  hookLoad(cargo) {
+    if (this.currentLoad === null && cargo.weight <= this.maxLoadWeight) {
+      this.currentLoad = cargo;
+    }
+
+    return this;
```
Comment (DenysKleimenov) [inline src/robotFactory.js:92]: we don't need to return the object

File: src/robotFactory.js [hunk 4]
```diff
@@ -1,10 +1,102 @@
...
+  }
+  unhookLoad() {
+    this.currentLoad = null;
+
+    return this;
```
Comment (DenysKleimenov) [inline src/robotFactory.js:97]: same here

**Review by DenysKleimenov — APPROVED** [review]
Change variable for step in the method for z coords

**Review by silchyn — CHANGES_REQUESTED** [review]
Task doesn’t meet all checklist requirements

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**